components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -50,20 +50,24 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
             _logger = logger;
             _identity = identity;
 
-            _messageTrigger.OnMessage += MessageTriggerMessageReceived;
-
             if (_config.BatchSize.HasValue && _config.BatchSize.Value > 1) {
                 _dataSetMessageBufferSize = _config.BatchSize.Value;
             }
             if (_config.MaxMessageSize.HasValue && _config.MaxMessageSize.Value > 0) {
                 _maxEncodedMessageSize = _config.MaxMessageSize.Value;
             }
+            
+            _diagnosticInterval = _config.DiagnosticsInterval.GetValueOrDefault(TimeSpan.MaxValue);
+            _batchTriggerInterval = _config.BatchTriggerInterval.GetValueOrDefault(TimeSpan.MaxValue);
+            _diagnosticsOutputTimer = new Timer(DiagnosticsOutputTimer_Elapsed, null,
```

Comment:
I think there is a constructor that just takes the callback and nothing else.

Reply:
right ...